### DISCUSSION TITLE
How can I find correct measurements.

### DISCUSSION THREAD

**Discussion by LightCannon** [decadenza/SimpleStereo#10] [Q&A]
I'm having a problem in finding distances between objects using the library. My procedure is this:

1. I'm doing stereo calibration on Matlab, and adding the parameters to Python&OpenCV code
2. I rectify the rig
3. I apply this rectification on the target Left and Right images
4. I get points from rectified Left and Right which are matched. So for img1 (left) I take a point (u_left, v_left), find its corresponding point u_right (same v_left) on img2(right) then get this pair (u_left, v_left, u_left-u_right). This last element is the disparity (d)
5. I do same as 4 for another point (I will measure distance, so need 2 points (u1,v1,d1) and (u2,v2,d2)
6. I get the point as 3D world coordinates (x,y,z) (in meters) for every (u,v,d) I have using
```
def get_3dpoint(u, v, d):
      A = np.array([[u, v, d, 1]]).T
      Q = self.get_Q()
      
      M = (Q).dot(A)
      M = 0.001*M/M[-1,0]
      return M[:-1,:].T
```
7. Get the Euclidian distance between both (x,y,z) I got.

Unfortunataly, I'm not getting right dimensions. First, am I right conceptually?

Regards.

**Comment by decadenza** [comment]
Not sure what self.get_Q() is taken from. The rest seems OK.

Anyway, it is not possible for me to verify without your data and code.
I suggest to create a minimum working example that shows the problem you are having and the results you are expecting.

**Reply by LightCannon** [reply]
Sorry, forgot to add it. This is a function I added inside RectifiedStereoRig class, it is the same as ur way to get Q, just separate function. Here is it.

```   
def get_Q(self):
    b   = self.getBaseline()
    fx  = self.K1[0,0]
    fy  = self.K2[1,1]
    cx1 = self.K1[0,2]
    cx2 = self.K2[0,2]
    a1  = self.K1[0,1]
    a2  = self.K2[0,1]
    cy  = self.K1[1,2]
    
    Q = np.eye(4, dtype='float64')
    
    Q[0,1] = -a1/fy
    Q[0,3] = a1*cy/fy - cx1
    
    Q[1,1] = fx/fy
    Q[1,3] = -cy*fx/fy
                            
    Q[2,2] = 0
    Q[2,3] = -fx
    
    Q[3,1] = (a2-a1)/(fy*b)
    Q[3,2] = 1/b                        
    Q[3,3] = ((a1-a2)*cy+(cx2-cx1)*fy)/(fy*b) 
    
    return Q
```

**Reply by LightCannon** [reply]
I will also attach my code and data soon, so you can take a look and tell me what can be wrong.
